pkg/operator/controller/gatewayclass/subscription.go

```diff
@@ -48,18 +53,19 @@ func (r *reconciler) ensureServiceMeshOperatorSubscription(ctx context.Context)
 }
 
 // desiredSubscription returns the desired subscription.
-func desiredSubscription(name types.NamespacedName) (*operatorsv1alpha1.Subscription, error) {
+func (r *reconciler) desiredSubscription(name types.NamespacedName) (*operatorsv1alpha1.Subscription, error) {
```

Comment:
Better to pass the required values as explicit parameters rather than passing in the whole reconciler as a receiver.  We want the `desired` functions not to depend on the client.